### PR TITLE
GEODE-5953: set target member when transaction hosted locally

### PR DESCRIPTION
**Pull request by pivotal-eshu**
* to be able to detect primary bucket has moved due to rebalance.
  * add debug level logging for transaction.

### CODE REVIEW

File: geode-core/src/distributedTest/java/org/apache/geode/internal/cache/partitioned/fixed/FixedPartitioningWithTransactionDistributedTest.java [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+      keySet.add(3);
+      ResultCollector resultCollector =
+          execution.withFilter(keySet).execute(new MyResumeTransactionFunction());
+      resultCollector.getResult();
+    } finally {
+      manager.rollback();
+    }
+  }
+
+  public static class MyFixedPartitionResolver implements FixedPartitionResolver {
```
Comment (kirklund) [inline geode-core/src/distributedTest/java/org/apache/geode/internal/cache/partitioned/fixed/FixedPartitioningWithTransactionDistributedTest.java]: I would change these inner classes to private if you can.

File: geode-core/src/main/java/org/apache/geode/internal/cache/TXManagerImpl.java [hunk 2]
```diff
@@ -340,6 +341,9 @@ public void begin() {
       proxy = new TXStateProxyImpl(cache, this, id, null);
     }
     setTXState(proxy);
+    if (logger.isDebugEnabled()) {
+      logger.debug("begin tx: " + proxy);
```
Comment (kirklund) [inline geode-core/src/main/java/org/apache/geode/internal/cache/TXManagerImpl.java]: Change log statements to use this syntax:
```
logger.debug("begin tx: {}", proxy);
```

File: geode-core/src/main/java/org/apache/geode/internal/cache/TXManagerImpl.java [hunk 3]
```diff
@@ -994,6 +1005,9 @@ public void masqueradeAs(TXStateProxy txState) {
       txState.getLock().lock();
     }
     setTXState(txState);
+    if (logger.isDebugEnabled()) {
+      logger.debug("masqueradeAs tx " + txState);
```
Comment (kirklund) [inline geode-core/src/main/java/org/apache/geode/internal/cache/TXManagerImpl.java]: Here's another that should change to use {}.

File: geode-core/src/distributedTest/java/org/apache/geode/internal/cache/partitioned/fixed/FixedPartitioningWithTransactionDistributedTest.java (hunk 1 above)
Comment (pivotal-eshu) [reply geode-core/src/distributedTest/java/org/apache/geode/internal/cache/partitioned/fixed/FixedPartitioningWithTransactionDistributedTest.java]: Have to change private to public due to following exception
Caused by: java.lang.NoSuchMethodException: org.apache.geode.internal.cache.partitioned.fixed.FixedPartitioningWithTransactionDistributedTest$MySuspendTransactionFunction

File: geode-core/src/main/java/org/apache/geode/internal/cache/TXManagerImpl.java (hunk 2 above)
Comment (pivotal-eshu) [reply geode-core/src/main/java/org/apache/geode/internal/cache/TXManagerImpl.java]: Will make the change.

**Pull request merged**